src/services/staff/staff-router.ts

```diff
@@ -77,6 +77,73 @@ staffRouter.post("/attendance/", strongJwtVerification, async (req: Request, res
     return res.status(StatusCode.SuccessOK).send({ status: "Success" });
 });
 
+/**
+ * @api {put} /staff/scan-attendee/ PUT /staff/scan-attendee/
+ * @apiGroup Staff
+ * @apiDescription Record user attendance for an event.
+ *
+ * @apiHeader {String} Authorization JWT Token with staff permissions.
+ *
+ * @apiBody {String} userId The attendee to check in.
+ * @apiBody {String} eventId The unique identifier of the event.
+ *
+ * @apiSuccessExample Example Success Response:
+ * HTTP/1.1 200 OK
+ * {success: true}
+ *
+ * @apiUse strongVerifyErrors
+ * @apiError (403: Forbidden) {String} InvalidPermission Access denied for invalid permission.
+ * @apiError (400: Bad Request) {String} InvalidParams Invalid or missing parameters.
+ * @apiError (400: Bad Request) {String} AlreadyCheckedIn Attendee has already been checked in for this event.
+ * @apiError (404: Not Found) {String} EventNotFound This event was not found
+ * @apiErrorExample Example Error Response:
+ *     HTTP/1.1 403 Forbidden
+ *     {"error": "Forbidden"}
+ * @apiErrorExample Example Error Response:
+ *     HTTP/1.1 400 Bad Request
+ *     {"error": "InvalidParams"}
+ * @apiErrorExample Example Error Response:
+ *     HTTP/1.1 400 Bad Request
+ *     {"error": "AlreadyCheckedIn"}
+ * @apiErrorExample Example Error Response:
+ *     HTTP/1.1 404 Not Found
+ *     {"error": "EventNotFound"}
+ */
+staffRouter.put("/scan-attendee/", strongJwtVerification, async (req: Request, res: Response, next: NextFunction) => {
+    const payload: JwtPayload | undefined = res.locals.payload as JwtPayload;
+    const userId: string | undefined = req.body.userId;
+    const eventId: string | undefined = req.body.eventId;
+
+    if (!userId || !eventId) {
+        return next(new RouterError(StatusCode.ClientErrorBadRequest, "InvalidParams"));
+    }
+
+    if (!hasStaffPerms(payload)) {
+        return next(new RouterError(StatusCode.ClientErrorForbidden, "Forbidden"));
+    }
+
+    const eventAttendance = await Models.EventAttendance.findOne({ eventId: eventId });
+
+    if (!eventAttendance) {
+        return next(new RouterError(StatusCode.ClientErrorNotFound, "EventNotFound"));
+    }
+
+    if (eventAttendance.attendees.includes(userId)) {
+        return next(new RouterError(StatusCode.ClientErrorBadRequest, "AlreadyCheckedIn"));
+    }
```

Comment:
Can you extract this into a helper function, and then use that in all attendance endpoints too?